### ISSUE TITLE
USS against z/OSMF doesn't work on Firefox

### ISSUE THREAD

**Issue by jackjia-ibm** [zowe/explorer-uss#27]
To request same api call `https://river.zowe.org:7554/api/v1/zosmf/restfiles/fs?path=/u/tstradm`, on Firefox it renders a request like this:

```
Host: river.zowe.org:7554
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:60.0) Gecko/20100101 Firefox/60.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Referer: https://river.zowe.org:7554/ui/v1/explorer-uss/
x-csrf-zosmf-header: *
origin: https://river.zowe.org:7554
Cookie: connect.sid=s%3Av3J7QBIrryQYmlaXw7Dw9ZMZ-ItJP6zF.VDe025x9chAk6JgCwV8YKOqCQOdpKKMPs%2F1MloOU6ms; LtpaToken2=ZOzTDLpp3xyIPKuBsN/p4Nw7y7k7vZZEz9mP+qJ4rxI+oIOZvTU/RLrLZZvPcSZ3AvNk5cBLMvspQapPSnPxmg9EnTd9FEShr8h5pjY4jfHHrzNsFBIG2GXFh0Fy5S4iRHOcf/fKr/W+MWRH/O7INULAmYP0OvFXAzfN0p/1T7Bkr57ZrHDR3rB2dVbewc3qUvGVYFSAd6hh5lVrcEMJqrIDfOp4njutNciPYfB8iEbR7FMvCjKATL8bywX/joF3Xy51D5e7pEPy03cNzeSVzF2IZfqo+SOr3tBw00aCRTHN82/TT2j8PTlNsxjUM9hJ
DNT: 1
Connection: keep-alive
```

And the response is:

```
HTTP/1.1 403 
X-Powered-By: Servlet/3.0
Cache-Control: no-store
Pragma: no-cache
Content-Language: en-US
Content-Type: application/json
Transfer-Encoding: chunked
Date: Tue, 08 Jan 2019 14:48:25 GMT

{ "errorID":"IZUG846W","errorMsg":"IZUG846W: An HTTP request for a z/OSMF REST service was received from a remote site. The request was rejected, however, because the remote site \"https://river.zowe.org:7554\" is not permitted to z/OSMF server \"IZUSVR\" on target system \"S0W1:10443\" ."}
```
, which is permission denied

While on Chrome, it renders a request like this:

```
GET /api/v1/zosmf/restfiles/fs?path=/u/tstradm HTTP/1.1
Host: river.zowe.org:7554
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
DNT: 1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36
X-CSRF-ZOSMF-HEADER: *
Accept: */*
Referer: https://river.zowe.org:7554/ui/v1/explorer-uss/
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,zh-CN;q=0.8,zh-HK;q=0.7,zh-TW;q=0.6,zh;q=0.5
Cookie: connect.sid=s%3ABtcoweg5T3VC7xSgwfE6YFFL6SYpOgta.wIESJ32ZTjaWk7bh9f4jl2r457sFI5XDI3Sj9hCTyps; LtpaToken2=ZOzTDLpp3xyIPKuBsN/p4Nw7y7k7vZZEz9mP+qJ4rxI+oIOZvTU/RLrLZZvPcSZ3AvNk5cBLMvspQapPSnPxmg9EnTd9FEShr8h5pjY4jfHHrzNsFBIG2GXFh0Fy5S4iRHOcf/fKr/W+MWRH/O7INULAmYP0OvFXAzfN0p/1T7Bkr57ZrHDR3rB2dVbewc3qUvGVYFSAd6hh5lVrcEMJqrIDfOp4njutNciPYfB8iEbR7FMvCjKATL8bywX/joF3Xy51D5e7pEPy03cNzeSVzF2IZfqo+SOr3tBw00aCRTHN82/TT2j8PTlNsxjUM9hJ
```

And the response is:

```
HTTP/1.1 200
X-Powered-By: Servlet/3.0
Cache-Control: no-store
Pragma: no-cache
X-IBM-Txid: tx000000000001DD35
Content-Language: en-US
Content-Type: application/json;charset=UTF-8
Transfer-Encoding: chunked
Date: Tue, 08 Jan 2019 14:48:49 GMT
```
, which is successful.

The difference is "origin:" header Firefox sent out. If I remove the origin header manually, the request will be successful.

**Comment by jackjia-ibm** [comment]
@Colin-Stone @stevenhorsman I just tested with Zowe build 425, the issue is still existing on Firefox.

**Comment by Colin-Stone** [comment]
I have only seen this problem when running in development mode from localhost, but it runs normally when the server is remote. I think you said your example is whilst running tests as part of the build. We need to keep tracking it and try and fix it but IMHO I would prioritize regressed functionality over it.

**Comment by jackjia-ibm** [comment]
@Colin-Stone this could be a zD&T special issue because the domain in zD&T box is S0W1 and it's different from what we accessing from outside.

Also I found it will happen on Chrome too in some cases. If I try to create directory, I got the same failure on Chrome and when checking the request, the "origin" header is added. But for Chrome, the GET requests like listing files in directory are working ok (without origin header added).
